File: C++/PAs/Matthews_Sebastian_PA2.cpp
/*  Author: Sebastian Matthews
    CPSC PA2: Codenames
    Sept 15, 2021
    Gonzaga University
    This program checks whether a user has valid credentials by having them enter one of five codenames, then the appropriate code number.
    Note: the codenames are Utah, Omaha, Gold, Juno, and Sword. The code number is 55.
*/

#include <iostream>
using namespace std;

int main(){
    
    // All declared variables needed for program are at the start of the code.
    string codeName;
    int codeNumber, amountOfGuesses = 0;

    // Introductory statement, followed by user input.
    cout << "Hello, stranger."<< endl;
    cout << "Enter a codename." << endl; 
    cin >> codeName;

    // The "if" below checks whether or not the inputted codename matches any of the five provided.
        // If it's true, it'll ask the user for the code number and "generate"(add 5 and multiply it by 7) a new one.
        //If it's false, the program will mark the number of tries and end when those tries are exceeded.
    if(codeName == "Utah" || codeName == "Omaha" || codeName == "Gold" || codeName == "Juno" || codeName == "Sword"){
            
            cout << "Good to see you again, " << codeName << "." << endl;
            cout << "One more thing, " << codeName << "..." << endl;
            cout << "What's the passcode?" << endl;
            cin >> codeNumber;
            
            if(codeNumber == 55){
                
                cout << "Correct! Identity confirmed!" << endl;
                cout << "Generating new key..." << endl;
                codeNumber += 5, codeNumber *= 7;
                cout << "Your new key is " << codeNumber << "." << endl;

            }
            else{
                
                cout << "Incorrect!" << endl;

            }

        }    
    else{
        
        cout << "Invalid codename." << endl; 
        amountOfGuesses = amountOfGuesses + 1;
        cout << "You have " << 3 - amountOfGuesses << " guesses remaining." << endl;
        cin >> codeName;
        
        if(codeName == "Utah" || codeName == "Omaha" || codeName == "Gold" || codeName == "Juno" || codeName == "Sword"){
            
            cout << "Good to see you again, " << codeName << "." << endl;
            cout << "One more thing, " << codeName << "..." << endl;
            cout << "What's the passcode?" << endl;
            cin >> codeNumber;
            
            if(codeNumber == 55){
                
                cout << "Correct! Identity confirmed!" << endl;
                cout << "Generating new key..." << endl;
                codeNumber += 5, codeNumber *= 7;
                cout << "Your new key is " << codeNumber << "." << endl;

            }
            else{
                
                cout << "Incorrect!" << endl;

            }

        }
        else{
            
            cout << "Invalid codename." << endl; 
            amountOfGuesses = amountOfGuesses + 1;
            cout << "You have " << 3 - amountOfGuesses << " guesses remaining." << endl;
            cin >> codeName;
            
            if(codeName == "Utah" || codeName == "Omaha" || codeName == "Gold" || codeName == "Juno" || codeName == "Sword"){
                
                cout << "Good to see you again, " << codeName << "." << endl;
                cout << "One more thing, " << codeName << "..." << endl;
                cout << "What's the passcode?" << endl;
                cin >> codeNumber;
                
                if(codeNumber == 55){
                    
                    cout << "Correct! Identity confirmed!" << endl;
                    cout << "Generating new key..." << endl;
                    codeNumber += 5, codeNumber *= 7;
                    cout << "Your new key is " << codeNumber << "." << endl;
                
                }
                else{
                    
                    cout << "Incorrect!" << endl;
                
                }
            }
            else{
                
                cout << "Invalid codename." << endl; 
                amountOfGuesses = amountOfGuesses + 1;
                cout << "You have " << 3 - amountOfGuesses << " guesses remaining." << endl;
                cin >> codeName;
                
                if(codeName == "Utah" || codeName == "Omaha" || codeName == "Gold" || codeName == "Juno" || codeName == "Sword"){
                    
                    cout << "Good to see you again, " << codeName << "." << endl;
                    cout << "One more thing, " << codeName << "..." << endl;
                    cout << "What's the passcode?" << endl;
                    cin >> codeNumber;
                    
                    if(codeNumber == 55){
                        
                        cout << "Correct! Identity confirmed!" << endl;
                        cout << "Generating new key..." << endl;
                        codeNumber += 5, codeNumber *= 7;
                        cout << "Your new key is " << codeNumber << "." << endl;
                    
                    }
                    else{
                        
                        cout << "Incorrect!" << endl;

                    }    
                }
                else{
                    
                    cout << "You have reached the maximum amount of guesses. Restarting..." << endl;
                    return 0;

                }

            }
            
            }     
        }
    }

File: Python/PAs/Matthews_Sebastian_PA2.py
#   Author: Sebastian Matthews
#   CPSC 215 PA2: Codename Confirmation
#   Jan 26, 2022
#   Gonzaga University
#   This program asks two users to input a codename, proceeding to "lock out" the user who inputted theirs incorrectly.
#   If both users input said codename correctly, both are granted access.

# Declare "constant"(in the sense that I won't change and emphasized by the macro-case) variables upfront.
CORRECT_CODE_NAME = 'sword'

def runInstance(): # This function controls the entire decision structure when called.
    for x in range (1,4): # This for loop lets the function run 3 times, which I looked up.
        if(x == 1):
            print('You start with 3 tries.') # Inform the users of their chances.
        inputName = input('Agent 1, enter codename.\n')
        inputOtherName = input('Agent 2, enter codename.\n')
        if(inputName == CORRECT_CODE_NAME and inputOtherName == CORRECT_CODE_NAME):
            print('Both codenames correct. Welcome, agents 1 and 2!')
            return 0
        elif(inputName != CORRECT_CODE_NAME and inputOtherName == CORRECT_CODE_NAME):
            if(x == 2):
                print('Agent 1\'s input is incorrect!\nRe-enter both codenames. You have', 3 - x, 'try remaining.')
            else:
                print('Agent 1\'s input is incorrect!')
                if(x != 3):
                    print('Re-enter both codenames. You have', 3 - x, 'tries remaining.')
        elif(inputName == CORRECT_CODE_NAME and inputOtherName != CORRECT_CODE_NAME):
            if(x == 2):
                print('Agent 2\'s input is incorrect!\nRe-enter both codenames. You have', 3 - x, 'try remaining.')    
            else:
                print('Agent 2\'s input is incorrect!')
                if(x != 3):
                    print('Re-enter both codenames. You have', 3 - x, 'tries remaining.')
        else:
            if(x == 2):
                print('Both incorrect!\nRe-enter both codenames. You have', 3 - x, 'try remaining.')
            else:
                print('Both incorrect!')
                if(x != 3):
                    print('Re-enter both codenames. You have', 3 - x, 'tries remaining.')

    # Below is a series of fail conditions, in which the program closes if met.
    if(x == 3 and inputName == CORRECT_CODE_NAME and inputOtherName != CORRECT_CODE_NAME):
        print('Agent 1\'s input is correct! Welcome, agent 1!\nAgent 2, you\'re locked out.')
    elif(x == 3 and inputName != CORRECT_CODE_NAME and inputOtherName == CORRECT_CODE_NAME):
        print('Agent 2\'s input is correct! Welcome, agent 2!\nAgent 1, you\'re locked out.')
    else:
        print('Both agents locked out! Program shutting down...')

def main():
    runInstance()

main() # Main has to be called, just like any other function.